packages/contracts-bedrock/test/properties/medusa/Protocol.properties.t.sol

```diff
@@ -87,7 +100,119 @@ contract ProtocolProperties is ProtocolHandler {
+        fromIndex = bound(fromIndex, 0, allSuperTokens.length - 1);
+        address recipient = getActorByRawIndex(recipientIndex);
+        OptimismSuperchainERC20 sourceToken = OptimismSuperchainERC20(allSuperTokens[fromIndex]);
+        OptimismSuperchainERC20 destinationToken =
+            MESSENGER.crossChainMessageReceiver(address(sourceToken), destinationChainId);
+        // TODO: perhaps the mock should already start ignoring this?
+        require(address(destinationToken) != address(0));
+        uint256 sourceBalanceBefore = sourceToken.balanceOf(currentActor());
+        uint256 sourceSupplyBefore = sourceToken.totalSupply();
+        uint256 destinationBalanceBefore = destinationToken.balanceOf(recipient);
+        uint256 destinationSupplyBefore = destinationToken.totalSupply();
+
+        vm.prank(currentActor());
+        try sourceToken.sendERC20(recipient, 0, destinationChainId) {
+            uint256 sourceBalanceAfter = sourceToken.balanceOf(currentActor());
+            uint256 destinationBalanceAfter = destinationToken.balanceOf(recipient);
+            assert(sourceBalanceBefore == sourceBalanceAfter);
+            assert(destinationBalanceBefore == destinationBalanceAfter);
+            uint256 sourceSupplyAfter = sourceToken.totalSupply();
+            uint256 destinationSupplyAfter = destinationToken.totalSupply();
+            assert(sourceSupplyBefore == sourceSupplyAfter);
+            assert(destinationSupplyBefore == destinationSupplyAfter);
+        } catch {
+            assert(address(destinationToken) == address(sourceToken));
+        }
+    }
+
+    /// @custom:property-id 9
+    /// @custom:property calls to relayERC20 with a value of zero dont modify accounting
+    function property_RelayZeroDoesNotModifyAccounting(
+        uint256 fromIndex,
+        uint256 recipientIndex
+    )
+        external
+        withActor(msg.sender)
+    {
+        fromIndex = bound(fromIndex, 0, allSuperTokens.length - 1);
+        address recipient = getActorByRawIndex(recipientIndex);
+        OptimismSuperchainERC20 token = OptimismSuperchainERC20(allSuperTokens[fromIndex]);
+        uint256 balanceBefore = token.balanceOf(currentActor());
+        uint256 supplyBefore = token.totalSupply();
+
+        MESSENGER.setCrossDomainMessageSender(address(token));
+        vm.prank(address(MESSENGER));
+        try token.relayERC20(currentActor(), recipient, 0) {
+            MESSENGER.setCrossDomainMessageSender(address(0));
+        } catch {
+            MESSENGER.setCrossDomainMessageSender(address(0));
+        }
+        uint256 balanceAfter = token.balanceOf(currentActor());
+        uint256 supplyAfter = token.totalSupply();
+        assert(balanceBefore == balanceAfter);
+        assert(supplyBefore == supplyAfter);
+    }
+
+    /// @custom:property-id 7
+    /// @custom:property calls to relayERC20 always succeed as long as the cross-domain caller is valid
```

Comment:
the success case isn't tested tho?
should the try have an assert sender==messenger (and catch the opposite), then fuzz the sender?

Reply:
good find :eyes: 

the `relayERC20` call, however, will mint tokens, which without non-atomic bridging will increase the total supply. I have two options for that
- track the increase in the appropiate ghost variables (which would kind of mean cheating on property 12)
- explicitly revert in the `try{}` block to discard the state transition

optimistically going for the latter